### PR TITLE
feat(nextjs): Improve debug logging

### LINKED ISSUE #1683: @clerk/nextjs logs are truncated on netlify
- [x] I have reviewed [the documentation](https://clerk.com/docs)
- [x] I have [searched for existing issues](https://github.com/clerkinc/javascript/issues)
- [x] I have ensured I am on the most recent version, and [checked the changelog](https://github.com/clerkinc/javascript/releases) for that version
- [x] I have provided the Frontend API key from my application dashboard: `pk_test_Y2hlZXJmdWwtbWFtbW90aC00OC5jbGVyay5hY2NvdW50cy5kZXYk`
- [ ] I have provided a minimal reproduction or replay recording below

### Brief Summary of the Issue
I'm unable to view the full output of the debug logs on netlify as netlify seems to cap the length of it. This is because the package emits all logs on one line instead of one per debug statement.

It would be great if this decision was either reconsidered, or a logger instance of our own could be passed in to workaround this problem. It seems to be a problem in vercel environments too considering the truncate function.

### Minimal Reproduction or Replay

> _We **require** either a minimal reproduction or a [replay recording](https://www.replay.io/) to be able to address a bug report. Any issues lacking a minimal reproduction or replay recording will be closed._

This [line in the debug logger](https://github.com/clerkinc/javascript/blob/d684e6f1f7913350bab0d9fb2424ca783b5a7993/packages/nextjs/src/utils/debugLogger.ts#L28) causes all log lines to be emitted in one `console.log`

I appreciate that netlify's handling is out of your control, but the decision to combine all logs into one call has made for an extremely frustrating debugging experience

A repro/recording here really wouldn't tell you anything more than the ticket/screenshot 

See the logs here are truncated before the headers have even finished printing (this is datadog, but it's the same in the netlify log streaming UI)
CleanShot 2023-09-05 at 16 55 13

Patching the package (using `pnpm patch @clerk/nextjs@4.23.0`) and to make each `.debug()` call trigger a `console.log` immediately allows me to inspect all logs, and finally see the auth state log section
CleanShot 2023-09-05 at 19 29 46

### Browser/OS
N/A

## Package + Version
- [x] `@clerk/nextjs` 4.23.0

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

Updates our debug logger to use individual `console.log` calls for each line. This is done to avoid platform log limitations per call. Along with this, I've improved the output of the logger slightly and added the relevant package versions to the log footer.

I did not add a unique ID to the output as the logs are still all logged together.

Output:
```
[clerk debug start: auth()]
  Headers debug, {
    "authStatus": "signed-in"
  }
  Options debug, {
    "apiKey": "",
    "secretKey": "sk_test_*********QDo",
    "apiUrl": "https://api.clerk.dev",
    "apiVersion": "v1",
    "authStatus": "signed-in"
  }
  JWT debug, <truncated>
[clerk debug end: auth()] (@clerk/nextjs=4.25.3,next=13.5.4)
```

fixes JS-716
fixes #1683 

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Review by octoper — APPROVED** [review]
💯

File: packages/nextjs/src/utils/debugLogger.ts [hunk 1]
```diff
@@ -77,6 +93,14 @@ export const withLogger: WithLogger = (loggerFactoryOrName, handlerCtor) => {
   }) as ReturnType<typeof handlerCtor>;
 };
 
+function debugLogHeader(name: string) {
+  return `[clerk debug start: ${name}]`;
+}
+
+function debugLogFooter(name: string) {
+  return `[clerk debug end: ${name}] (@clerk/nextjs=${PACKAGE_VERSION},next=${nextPkg.version})`;
```
Comment (dimkl) [inline packages/nextjs/src/utils/debugLogger.ts:101]: 🎉

**Review by dimkl — APPROVED** [review]
:cloud I think we should add a ticket to handle the logging using a logger adapter to provide and use a default logger adapter that will handle truncating for vercel, single-line logs, ...

File: packages/nextjs/src/utils/debugLogger.test.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ describe('withLogger', () => {
       logger.debug(veryLongString);
     });
     handler();
-    expect(log.mock.calls[0][0]).toHaveLength(4096);
+    expect(log.mock.calls[1][0]).toHaveLength(4096);
```
[the remark this answers is not in the document]
Comment (BRKalow) [reply packages/nextjs/src/utils/debugLogger.test.ts:128]: good idea 👍

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**